### PR TITLE
Add wdio spec to test pipes-shell

### PR DESCRIPTION
**Pull request by sjmiles**
also
- some refactoring to support the tests
- and fix bugs that had crept up in the absence of tests

### CODE REVIEW

**Comment by sjmiles** [comment]
cc @ianull just for reference, this test designed to prevent a bug that bit him

**Comment by ianull** [comment]
Thanks Scott for doing this.

File: shells/pipes-shell/source/testDevice.js [hunk 1]
```diff
@@ -48,29 +90,28 @@ export const smokeTest = async (paths, storage, manifest, bus) => {
...
   const notificationTest = () => {
     // spawn an arc
-    send({message: 'spawn', modality: 'dom', recipe: 'Notification'});
+    send({message: 'runArc', arcId: 'pipe-notification-test', modality: 'dom', recipe: 'Notification'});
   };
   //
   const parseTest = () => {
     // parse manifest content
-    send({message: 'parse', content: `import 'https://$particles/canonical.arcs'`});
+    send({message: 'parse', content: `import 'https://$particles/PipeApps/PipeApps.arcs'`});
```
Comment (mmandlis) [inline shells/pipes-shell/source/testDevice.js]: just a question - is the PipeApps/canonical.arcs ever used, or should it be removed?

Comment (sjmiles) [reply shells/pipes-shell/source/testDevice.js]: I have been in the habit of naming a default manifest `<folder-name>.arcs` (if I make one). Its helpful if the manifest identifies itself specifically.

`canonical.arcs` can be removed, unless the above is wrong-think.

**Pull request merged**